presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeGeneratorContext.java

```diff
@@ -82,17 +84,29 @@ public FunctionRegistry getRegistry()
         return registry;
     }
 
+    public BytecodeNode generateCall(String name, ScalarFunctionImplementation function, List<BytecodeNode> arguments)
+    {
+        return generateCall(name, function, arguments, Optional.empty(), Optional.empty());
+    }
+
     /**
      * Generates a function call with null handling, automatic binding of session parameter, etc.
      */
-    public BytecodeNode generateCall(String name, ScalarFunctionImplementation function, List<BytecodeNode> arguments)
+    public BytecodeNode generateCall(
+            String name,
+            ScalarFunctionImplementation function,
+            List<BytecodeNode> arguments,
+            Optional<Variable> outputBlock,
+            Optional<Type> outputType)
     {
+        checkArgument((!outputBlock.isPresent()) || outputType.isPresent(), "outputType must present if outputBlock is present");
```

Comment:
Maybe a better way to write the error message is something like: `outputBlock should not be present when the SQL function does not have return value`